R/module_teal.R

```diff
@@ -184,33 +187,81 @@ srv_teal <- function(id, data, modules, filter = teal_slices()) {
       }
     )
 
-    data_rv <- srv_init_data("data", data = data, modules = modules, filter = filter)
+    data_pulled <- srv_init_data("data", data = data)
+    data_validated <- srv_validate_reactive_teal_data(
+      "validate",
+      data = data_pulled,
+      modules = modules,
+      validate_shiny_silent_error = FALSE
+    )
+    data_init <- reactive({
+      req(inherits(data_validated(), "teal_data"))
+      is_filter_ok <- check_filter_datanames(filter, .teal_data_ls(data_validated()))
+      if (!isTRUE(is_filter_ok)) {
+        showNotification(
+          "Some filters were not applied because of incompatibility with data. Contact app developer.",
+          type = "warning",
+          duration = 10
+        )
+        warning(is_filter_ok)
+      }
+      .add_signature_to_data(data_validated())
+    })
+
+    status <- reactive({
+      if (inherits(data_pulled(), "teal_data")) {
+        "ok"
+      } else if (inherits(data, "teal_data_module")) {
+        "disable"
+      } else {
+        "hide"
+      }
+    })
```

Comment:
The only thing that is a bit confusing, and the thing I need to rethink once more is why we use `data_pulled()` in here, where we also have `data_validated` and `data_init` build upon it. I guess instead of `data_pulled()` we could use `data_init()`?

Reply:
- `data_pulled` can return error
- `data_validated` always returns `teal_data` (after being NULL initially)
- `data_init` is `teal_data` with hashes in the `@code` (`.add_signature_to_data` )